enhancements/machine-config/pin-and-pre-load-images.md

```diff
@@ -244,89 +273,40 @@ complete OpenShift releases: blobs take 16 GiB and when they are written to disk
 they take 32 GiB.
 
 If the calculate disk space exceeds the available disk space then the
-`PinnedImageSet` will be marked as failed and the process will not move forward.
-For example:
-
-```yaml
-status:
-  conditions:
-  - type: Ready
-    status: "False"
-  - type: Failed
-    status: "True"
-    message: |
-      Pulling the images in `node12` requires at least 16 GiB of disk
-      space, but there are only 10 GiB available
-```
+`pinned_image_manager` will report the error as a `Degraded=true` status
+for the `MachineConfigPool` with the error in the `message`.
 
 Note that even with this check it will still be possible (but less likely) to
 have failures to pull images due to disk space: the heuristic could be wrong,
 and there may be other components pulling images or consuming disk space in
 some other way. Those failures will be detected and reported in the status of
-the `PinnedImageSet` when CRI-O fails to pull the image.
+the `MachineConfigPool` when CRI-O fails to pull the image.
 
-The steps above will happen in all the nodes of the cluster. The daemon set
-will be provided with enough information to ensure that each pod applies only
-the changes required for the node where it runs, according to the node
-selectors in the `PinnedImageSet` custom resources.
+Once all the relevant images are successfully pinned and downloaded to the
+matching nodes, the `pinned_image_manager` will signal the completion of the
+process by invoking nodeWriter.SetDone(). This action notifies the
```

Comment:
Maybe I am overthinking. Calling nodeWriter.SetDone() would mean that we are creating two sources where  anode is marked as update completed, so we need a way to ensure that we don't do that while MCD is also performing an update due to rendered-config change. Probably, this isn't an issue for targeted usecase but making this a general feature would require better thought process.

Reply:
I will reword there is just a single sync of MCD so `Done` will only be applied by MCD. I was trying to convey that no additional mechanisms were required.  But how that will work was not described will flesh that out.

Reply:
I have a similar question here, I guess we're sort of introducing an inter-dependency between the "machineconfig update" and the "pinnedimageset update" which, reading the controller code, shouldn't conflict unless there is a degrade. But this does sort of raise some clarifications such as:

1. do we want one's failure to block the other
2. if you roll out one pinnedimageset and then try to roll out a second one (i.e. append another set to the list in the MCP) should we be waiting for the first to finish
3. if the daemon is working on a node, should it not reboot the node unless pinnedimageset is able to finish
4. does the `done` state otherwise affect node updates
5. how does the pinnedimageset controller actually know if the `done` state is reacting to the current set of pinnedimageset changes vs, say, the daemon is dead and hasn't reacted to the latest request yet?